src/DistributedWriteAheadLog/Record.cpp

```diff
@@ -3,31 +3,46 @@
 #include <DataStreams/NativeBlockInputStream.h>
 #include <DataStreams/NativeBlockOutputStream.h>
 /// #include <DataStreams/materializeBlock.h>
+#include <Compression/CompressedReadBuffer.h>
+#include <Compression/CompressedWriteBuffer.h>
+#include <Compression/CompressionFactory.h>
 #include <IO/ReadBufferFromMemory.h>
 #include <IO/ReadHelpers.h>
 #include <IO/WriteBufferFromVector.h>
 #include <IO/WriteHelpers.h>
 
 namespace DWAL
 {
-ByteVector Record::write(const Record & record)
+ByteVector Record::write(const Record & record, bool compressed)
 {
     ByteVector data{static_cast<size_t>((record.block.bytes() + 2) * 1.5)};
     DB::WriteBufferFromVector wb{data};
-    DB::NativeBlockOutputStream output(wb, 0, DB::Block{});
 
     /// Write flags
     /// flags bits distribution
     /// [0-4] : Version
     /// [5-10] : OpCode
-    /// [11-63] : Reserved
-    uint64_t flags = VERSION | (static_cast<UInt8>(record.op_code) << 5ul);
+    /// [11-13] : Compression
```

Comment:
If the codec is already self-contained in the binary data, we don't need reserve 2 bits for the compression codec ?